django_tenants/middleware/main.py

```diff
@@ -20,14 +23,37 @@ class TenantMainMiddleware(MiddlewareMixin):
 
     @staticmethod
     def hostname_from_request(request):
-        """ Extracts hostname from request. Used for custom requests filtering.
-            By default removes the request's port and common prefixes.
+        """Extracts hostname from request. Used for custom requests filtering.
+        By default removes the request's port and common prefixes.
         """
         return remove_www(request.get_host().split(':')[0])
 
+    @staticmethod
+    def get_cached_tenant(domain_model, hostname):
+        """Get cached tenant."""
+        if hostname in TENANTS_CACHE_DATA:
+            tenant = TENANTS_CACHE_DATA[hostname]
+            client_model = get_tenant_model()
+            if isinstance(tenant, client_model):
+                return tenant
+            else:
+                raise domain_model.DoesNotExist()
+        else:
+            try:
+                domain = domain_model.objects.select_related('tenant').get(domain=hostname)
+            except domain_model.DoesNotExist:
+                TENANTS_CACHE_DATA[hostname] = None
```

Comment:
Don't assign anything here aka remove this line and then you won't need the if condition on line 37, no ?